File: 890-find-and-replace-pattern/890-find-and-replace-pattern.py
class Solution:
    
    def map_word(self, word):
        _map = {}
        for i, char in enumerate(word):
            if char in _map:
                _map[char].append(i)
            else:
                _map[char] = [i]
        return list(_map.values())

    def findAndReplacePattern(self, words: List[str], pattern: str) -> List[str]:
        
        pattern_list = self.map_word(pattern)
        
        match = []
        for word in words:
            temp = self.map_word(word)
            if temp == pattern_list:
                match.append(word)
        return match

File: 890-find-and-replace-pattern/890-find-and-replace-pattern.java
class Solution {
    
    private boolean matchString(String a, String b){
        for (int i=0; i<a.length(); i++) {
            if(a.indexOf(a.charAt(i)) != b.indexOf(b.charAt(i))) return false;
        }
        return true;
    }
    public List<String> findAndReplacePattern(String[] words, String pattern) {
        List<String> match = new ArrayList<String>();
        
        for (String word: words) {
            if (matchString(word, pattern)) match.add(word);
        }
        
        return match;
    }
}